src/heretic/model.py

```diff
@@ -175,20 +353,20 @@ def try_add(component: str, matrix: Any):
 
         # Granite MoE Hybrid - attention layers with shared_mlp.
         with suppress(Exception):
-            try_add("mlp.down_proj", layer.shared_mlp.output_linear.weight)
+            try_add("mlp.down_proj", layer.shared_mlp.output_linear)
 
         # Granite MoE Hybrid - MoE layers with experts.
         with suppress(Exception):
             for expert in layer.moe.experts:
-                try_add("mlp.down_proj", expert.output_linear.weight)
+                try_add("mlp.down_proj", expert.output_linear)
 
         # We need at least one MLP down-projection.
-        assert matrices["mlp.down_proj"]
+        assert modules["mlp.down_proj"]
```

Comment:
This assertion will have to be modified, else it will fail for GPT-OSS. What we should probably check for instead is that there is at least one module in total, across all components.